python/lsst/meas/extensions/scarlet/io.py

```diff
@@ -0,0 +1,808 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+import json
+from typing import Any
+import numpy as np
+import warnings
+from scarlet.bbox import Box, overlapped_slices
+from scarlet.lite import LiteBlend, LiteFactorizedComponent, LiteObservation, LiteSource, LiteParameter
+from scarlet.lite.measure import weight_sources
+
+from lsst.geom import Box2I, Extent2I, Point2I, Point2D
+from lsst.afw.detection.multiband import heavyFootprintToImage
+
+from .source import liteModelToHeavy
+
+
+@dataclass
+class ScarletComponentData:
+    """Data for a component expressed as a 3D data cube
+
+    For now this is used for scarlet main source models because
+    their structure is too complex to persist in the same
+    way that scarlet lite components can be persisted.
+
+    Note that both `xy0` and `extent` use lsst ``(x, y)`` convention,
+    not the scarlet/C++ ``(y, x)`` convention.
```

Comment:
Thanks, but I think that in this case it makes sense because `xy0` (not `yxo`) and `extent` are LSST definitions and the `center` is ` Point2I`, so it makes sense to use that convention. Even if I used the opposite, I would still contain a line in the docstring to avoid potential confusion.